parquet-column/src/main/java/org/apache/parquet/column/values/bytestreamsplit/ByteStreamSplitValuesReaderForDouble.java

```diff
@@ -18,20 +18,15 @@
  */
 package org.apache.parquet.column.values.bytestreamsplit;
 
-import org.apache.parquet.bytes.BytesUtils;
+import static org.apache.parquet.bytes.BytesUtils.bytesToLong;
 
 public class ByteStreamSplitValuesReaderForDouble extends ByteStreamSplitValuesReader {
-
-  private final byte[] valueByteBuffer;
-
   public ByteStreamSplitValuesReaderForDouble() {
     super(Double.BYTES);
-    valueByteBuffer = new byte[Double.BYTES];
   }
 
   @Override
   public double readDouble() {
-    gatherElementDataFromStreams(valueByteBuffer);
-    return Double.longBitsToDouble(BytesUtils.bytesToLong(valueByteBuffer));
+    return Double.longBitsToDouble(bytesToLong(readElement().getBytesUnsafe()));
```

Comment:
`getBytesUnsafe` is doing a copy here, but to a very small array. I don't know enough about JVM performance to decide if this is a problem or not. @gszadovszky Do you have any idea or suggestion?

(the underlying problem is that `ByteBuffer` and `Binary` have different limitations in their respective APIs, and whether we want to use one or the other depends on which data type we want to convert to :-))

Reply:
I may just use both a backing `Binary` and a `ByteBuffer` pointing to it.

Reply:
(note to self: https://spotcodereviews.com/articles/optimization/2021/03/15/does-the-jvm-eliminate-allocations-of-temporary-objects.html has an interesting discussion of array allocation elision)